src/frontend/src/modals/IOModal/components/chatView/newChatView.tsx

```diff
@@ -0,0 +1,278 @@
+
+  const inputTypes = inputs.map((obj) => obj.type);
+  const updateFlowPool = useFlowStore((state) => state.updateFlowPool);
+  const [id, setId] = useState<string>("");
+  const { mutate: mutateDeleteFlowPool } = useDeleteBuilds();
+  const maxFileSizeUpload = useUtilityStore((state) => state.maxFileSizeUpload);
+
+  //build chat history
+  useEffect(() => {
+    const messagesFromMessagesStore: ChatMessageType[] = messages
+      .filter(
+        (message) =>
+          message.flow_id === currentFlowId &&
+          (visibleSession === message.session_id ?? true),
+      )
+      .map((message) => {
+        let files = message.files;
+        //HANDLE THE "[]" case
+        if (typeof files === "string") {
+          files = JSON.parse(files);
+        }
+        return {
+          isSend: message.sender === "User",
+          message: message.text,
+          sender_name: message.sender_name,
+          files: files,
+          id: message.id,
+          timestamp: message.timestamp,
+          session: message.session_id,
+          edit: message.edit,
+        };
+      });
+    const finalChatHistory = [...messagesFromMessagesStore].sort((a, b) => {
+      return new Date(a.timestamp).getTime() - new Date(b.timestamp).getTime();
+    });
+
+    setChatHistory(finalChatHistory);
+  }, [flowPool, messages, visibleSession]);
+  useEffect(() => {
+    if (messagesRef.current) {
+      messagesRef.current.scrollTop = messagesRef.current.scrollHeight;
+    }
+  }, []);
+
+  const ref = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (ref.current) {
+      ref.current.focus();
+    }
+    // trigger focus on chat when new session is set
+  }, [focusChat]);
+
+  function clearChat(): void {
+    setChatHistory([]);
+
+    mutateDeleteFlowPool(
+      { flowId: currentFlowId },
+      {
+        onSuccess: () => {
+          CleanFlowPool();
+        },
+      },
+    );
+    //TODO tell backend to clear chat session
+    if (lockChat) setLockChat(false);
+  }
+
+  function updateChat(
+    chat: ChatMessageType,
+    message: string,
+    stream_url?: string,
+  ) {
+    chat.message = message;
+    if (chat.componentId)
+      updateFlowPool(chat.componentId, {
+        message,
+        sender_name: chat.sender_name ?? "Bot",
+        sender: chat.isSend ? "User" : "Machine",
+      });
+  }
+  const [files, setFiles] = useState<FilePreviewType[]>([]);
+  const [isDragging, setIsDragging] = useState(false);
+
+  const { dragOver, dragEnter, dragLeave } = useDragAndDrop(setIsDragging);
+
+  const onDrop = (e) => {
+    e.preventDefault();
+    if (e.dataTransfer.files && e.dataTransfer.files.length > 0) {
+      handleFiles(e.dataTransfer.files, setFiles, currentFlowId, setErrorData);
+      e.dataTransfer.clearData();
+    }
+    setIsDragging(false);
+  };
+
+  const { mutate } = usePostUploadFile();
+
+  const handleFiles = (files, setFiles, currentFlowId, setErrorData) => {
```

Comment:
**just a question!**
can we create a separate custom hook file to handle this, just to clean up the code in the component?